src/EditProfile.js

```diff
@@ -0,0 +1,213 @@
+import React from "react";
+import "./profile.css"
+import {withRouter, Link} from 'react-router-dom'
+
+  class EditProfile extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        userId: this.props?.location?.state?.id,
+        name: "",
+        age: "null",
```

Comment:
I think the default for age is now 0 instead of "null", you can change that if you want.